src/sql/workbench/api/common/extHostModelView.ts

```diff
@@ -1602,6 +1602,16 @@ class DeclarativeTableWrapper extends ComponentWrapper implements azdata.Declara
 		this._proxy.$doAction(this._handle, this._id, ModelViewAction.Filter, rowIndexes);
 	}
 
+	public get selectedRow(): number {
+		return this.properties['selectedRow'] ?? -1;
+	}
+
+	public set selectedRow(v: number) {
+		this.setProperty('selectedRow', v);
```

Comment:
So now is this going to result in the event being fired twice - once when it's set on the ext host side and then once again after it's updated on the main thread side which then presumably causes the ext host to also be updated and thus fire the event? Or is the logic in onRowSelected handling that case?

It might be better to have selectedRow be read only and then have a separate function `selectRow(row: number)` that acts as the setter to avoid that.

Reply:
calling this setter won't cause the event to fire twice, it will only be fired from main thread when user manually select a row.

Reply:
I just move this event firing logic to main thread